setup: Fix ubuntu.sh install script

julianoes opened the pull request:
I'm taking a passs over the ubuntu setup script.

This fixes https://github.com/PX4/Devguide/issues/917.

dagar commented:
Add `psutil` for mavsdk tests? https://github.com/PX4/Firmware/blob/fb0bf267870104ddc1cf9b15bc23aa01872aac0e/.github/workflows/sitl_tests.yml#L16

dagar commented on Tools/setup/ubuntu.sh:
```diff
@@ -115,15 +116,15 @@ fi
 echo "Installing PX4 Python3 dependencies"
-sudo python3 -m pip install --upgrade pip setuptools wheel
-sudo python3 -m pip install -r ${DIR}/requirements.txt
+pip3 install --user --upgrade setuptools wheel
```
PATH update if we're going to go with the `--user` install? I know this was always the "right" way to do it, but it always seemed to cause more confusion then just installing system wide.

julianoes replied:
I've always done it this way and don't remember having problems. Which tools do we need from the user directory that require the path to be added?

dagar replied:
I'll try again on a fresh ubuntu setup. In the past I recall needing to set the PYTHONPATH to find things installed with `--user`. The other question is what happens in docker when this is run as root.

hamishwillee commented:
So this is what I get running setup script: https://gist.github.com/hamishwillee/7748b4ab60e556b835488c3a6e6e37a5

That has the errors from my comment above.

When I run same script again in same computer seems to no longer report those errors: https://gist.github.com/hamishwillee/b1e322d2202eb175077a073f88470134

Note the final message `Command 'arm-none-eabi-gcc' not found, but can be installed with:` is probably false/odd. The compiler isn't detected because we haven't rebooted yet.

When I try run jmavsim I end up with 
```
* SYS_AUTOCONFIG: curr: 1 -> new: 0
INFO  [dataman] Unknown restart, data manager file './dataman' size is 11798680 bytes
INFO  [simulator] Waiting for simulator to connect on TCP port 4560

```
So this didn't "work".

julianoes commented:
> @dagar: Add psutil for mavsdk tests? 

It's already listed in `requirements.txt`.

julianoes commented:
@hamishwillee this is now ready to be tested.

hamishwillee commented:
Hi @julianoes ,
Tried this again on ubuntu 18.04 (`source`d it) then switched to master. I'll have to retest but initial comments:

1. Failed with message on build

```
/tmp/headers -q
python import error:  No module named 'genmsg'

Required python packages not installed.

On a Debian/Ubuntu system please run:

  sudo apt-get install python3-empy
  sudo pip3 install pyros-genmsg

On MacOS please run:
  sudo pip3 install empy pyros-genmsg

On Windows please run:
  easy_install empy pyros-genmsg

[6/682] Generating parameters.xml
ninja: build stopped: subcommand failed.
Makefile:193: recipe for target 'px4_sitl_default' failed
make: *** [px4_sitl_default] Error 1
```

2. I had to acknowledge prompts 3 or 4 times. Better if it skips though.

3. It updated all these (if can be avoided would be good):
```
The following packages will be upgraded:
  libreoffice-avmedia-backend-gstreamer libreoffice-base-core libreoffice-calc
  libreoffice-common libreoffice-core libreoffice-draw libreoffice-gnome
  libreoffice-gtk3 libreoffice-impress libreoffice-math libreoffice-ogltrans
  libreoffice-pdfimport libreoffice-style-breeze libreoffice-style-galaxy
  libreoffice-style-tango libreoffice-writer python3-uno
```

julianoes commented:
Thanks for testing.

> 2. If I install sudo pip3 install pyros-genmsg then jmavsim will build.

That's indeed missing from requirements.txt. It's odd that it did not show in my testing in docker where nothing is pre-installed! I've added it now.

> 3. I had to acknowledge prompts 3 or 4 times. Better if it skips though.

What prompts? I've added `DEBIAN_FRONTEND=noninteractive` in every `apt-get install` command now, hopefully that fixes  it.

> 4. It updated all these (if can be avoided would be good):

I assume this comes with Python 3. I'm not sure if we can prevent it. Also, your system should generally be kept up-to-date anyway, I don't think that should be our problem.

MaEtUgR commented:
@julianoes Could you update the arch.sh script with the parts that apply there as well? I'm sorry for any duplication there. I was considering to make a common parts script that gets called.

julianoes commented:
I can do that later. Let's merge this for now.

LorenzMeier commented:
Thanks! Perfect is indeed the enemy of good.
